packages/go_router/example/lib/stateful_nested_navigation.dart

```diff
@@ -0,0 +1,420 @@
+                      label: 'B',
+                      detailsPath: '/b/details/1',
+                      secondDetailsPath: '/b/details/2'),
+              routes: <RouteBase>[
+                GoRoute(
+                  path: 'details/:param',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      DetailsScreen(label: 'B', param: state.params['param']),
+                ),
+              ],
+            ),
+          ),
+
+          /// The route branch for the third tab of the bottom navigation bar.
+          ShellRouteBranch(
+            /// Since this route branch has a nested StatefulShellRoute as the
+            /// root route, we need to specify what the default location for the
+            /// branch is.
+            defaultLocation: '/c1',
+            rootRoute: StatefulShellRoute.rootRoutes(
+              /// This bottom tab uses a nested shell, wrapping sub routes in a
+              /// top TabBar. In this case, we're using the `rootRoutes`
+              /// convenience constructor.
+              routes: <GoRoute>[
+                GoRoute(
+                  path: '/c1',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      const TabScreen(label: 'C1', detailsPath: '/c1/details'),
+                  routes: <RouteBase>[
+                    GoRoute(
+                      path: 'details',
+                      builder: (BuildContext context, GoRouterState state) =>
+                          const DetailsScreen(label: 'C1', withScaffold: false),
+                    ),
+                  ],
+                ),
+                GoRoute(
+                  path: '/c2',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      const TabScreen(label: 'C2', detailsPath: '/c2/details'),
+                  routes: <RouteBase>[
+                    GoRoute(
+                      path: 'details',
+                      builder: (BuildContext context, GoRouterState state) =>
+                          const DetailsScreen(label: 'C2', withScaffold: false),
+                    ),
+                  ],
+                ),
+              ],
+              builder: (BuildContext context, GoRouterState state,
+                  Widget ignoredNavigatorContainer) {
+                /// For this nested StatefulShellRoute we are using a custom
+                /// container (TabBarView) for the branch navigators, and thus
+                /// ignoring the default navigator contained passed to the
+                /// builder. Custom implementation can access the branch
+                /// navigators via the StatefulShellRouteState
+                /// (see TabbedRootScreen for details).
+                return const TabbedRootScreen();
+              },
+            ),
+          ),
+        ],
+        builder: (BuildContext context, GoRouterState state,
+            Widget navigatorContainer) {
```

Comment:
Consider renaming `navigatorContainer` to `child` to align with the `ShellRoute` examples

Reply:
👍